Wikipedia/Code/WMFHomeViewController.m

```diff
@@ -472,6 +466,37 @@ - (void)didTapFooterInSection:(NSUInteger)section {
     [self.navigationController pushViewController:extendedList animated:YES];
 }
 
+- (void)didTapHeaderInSection:(NSUInteger)section {
+    WMFHomeSection* homeSection = self.schemaManager.sections[section];
+    switch (homeSection.type) {
+        case WMFHomeSectionTypeContinueReading:
+        case WMFHomeSectionTypeMainPage:
+        case WMFHomeSectionTypeFeaturedArticle:
+        case WMFHomeSectionTypePictureOfTheDay:
+        case WMFHomeSectionTypeRandom:
+            [self selectFirstRowInSection:section];
+            break;
+        case WMFHomeSectionTypeNearby:
+            [self didTapFooterInSection:section];
+            break;
+        case WMFHomeSectionTypeSaved: {
+            WMFRelatedSectionController* controller = (WMFRelatedSectionController*)[self sectionControllerForSectionAtIndex:section];
+            [self wmf_pushArticleViewControllerWithTitle:controller.title
+                                         discoveryMethod:MWKHistoryDiscoveryMethodLink
+                                               dataStore:self.dataStore];
+            break;
```

Comment:
turns out we don't have a best practice on this, but related to my comment above, we should (in general) avoid `default` cases in our `switch` statements, unless it's actually the intention to have a global fallback.  in this case, we should remove it and force that all cases (now and future) are listed in the `switch`.